learn-ocaml.el

```diff
@@ -774,52 +819,122 @@ Note: this function will be used by `learn-ocaml-on-load-aux'."
           nil
           callback))))))
 
-(defun learn-ocaml-on-load-aux (token server callback)
+(cl-defun learn-ocaml-init-cmd (&key token server nickname secret callback)
+  "Run \"learn-ocaml-client init\" with options."
+  (learn-ocaml-print-time-stamp)
+  (learn-ocaml-make-process-wrapper
+   :name "init"
+   :command (learn-ocaml-command-constructor
+             :token token
+             :server server
+             :param1 nickname
+             :param2 secret
+             :command "init")
+   :stderr (learn-ocaml-log-buffer)
+   :sentinel (apply-partially
+              #'learn-ocaml-error-handler
+              nil
+              callback)))
+
+
+
+(defun learn-ocaml-connection (server callback)
+  "Connect the user with a (login,passwd) or a token and continue with the callback"
```

Comment:
```suggestion
  "Connect the user with a (login, passwd) or a token and continue with the CALLBACK"
```
(when you refer to variables in the elisp docstrings, they need to be in UPPERCASE)

If I'm not mistaken, learn-ocaml-connection is only used when use_passwd is enabled?

In this case, one should probably add this information in the docstring, and maybe change the name of the function, as well as renaming learn-ocaml-on-load-aux ?

For example:

* learn-ocaml-connection → learn-ocaml-login-possibly-with-passwd
* learn-ocaml-on-load-aux → learn-ocaml-login-with-token

What do you think?